src/QMCWaveFunctions/tests/test_rotatedspos.cpp

```diff
@@ -0,0 +1,305 @@
+
+  // Sanity check for orbs. Expect 2 electrons, 8 orbitals, & 79507 coefs/orb.
+  const auto orbitalsetsize = rot_spo->getOrbitalSetSize();
+  app_log() << "RotatedSPOs::OrbitalSetSize= " << orbitalsetsize << "\n";
+  REQUIRE(orbitalsetsize == 8);
+
+  // 2.) Get data for unrotated orbitals. Check that there's no rotation
+  rot_spo->buildOptVariables(elec_.R.size());
+  SPOSet::ValueMatrix psiM_bare(elec_.R.size(), orbitalsetsize);
+  SPOSet::GradMatrix dpsiM_bare(elec_.R.size(), orbitalsetsize);
+  SPOSet::ValueMatrix d2psiM_bare(elec_.R.size(), orbitalsetsize);
+  rot_spo->evaluate_notranspose(elec_, 0, elec_.R.size(), psiM_bare, dpsiM_bare, d2psiM_bare);
+
+  // This stuff checks that no rotation was applied. See test_einset.cpp
+  // value
+  REQUIRE(std::real(psiM_bare[1][0]) == Approx(-0.8886948824));
+  REQUIRE(std::real(psiM_bare[1][1]) == Approx(1.4194120169));
+  // grad
+  REQUIRE(std::real(dpsiM_bare[1][0][0]) == Approx(-0.0000183403));
+  REQUIRE(std::real(dpsiM_bare[1][0][1]) == Approx(0.1655139178));
+  REQUIRE(std::real(dpsiM_bare[1][0][2]) == Approx(-0.0000193077));
+  REQUIRE(std::real(dpsiM_bare[1][1][0]) == Approx(-1.3131694794));
+  REQUIRE(std::real(dpsiM_bare[1][1][1]) == Approx(-1.1174004078));
+  REQUIRE(std::real(dpsiM_bare[1][1][2]) == Approx(-0.8462534547));
+  // lapl
+  REQUIRE(std::real(d2psiM_bare[1][0]) == Approx(1.3313053846));
+  REQUIRE(std::real(d2psiM_bare[1][1]) == Approx(-4.712583065));
+
+  /* 
+     3.) Apply a rotation to the orbitals
+         To do this, construct a params vector and call the
+	 RotatedSPOs::apply_rotation(params) method. That should do the 
+	 right thing for this particular spline class.
+	 
+	 For 2 electrons in 8 orbs, we expect 2*(8-2) =12 params.
+  */
+  const auto rot_size = rot_spo->m_act_rot_inds.size();
+  REQUIRE(rot_size == 12); // = Nelec*(Norbs - Nelec) = 2*(8-2) = 2*6 = 12
+  std::vector<double> param(rot_size);
+  for (auto i = 0; i < rot_size; i++)
+  {
+    param[i] = 0.01 * i;
+  }
+  rot_spo->apply_rotation(param, false); // Expect this to call SplineR2R::applyRotation()
+
+  // 4.) Get data for rotated orbitals.
+  SPOSet::ValueMatrix psiM_rot(elec_.R.size(), orbitalsetsize);
+  SPOSet::GradMatrix dpsiM_rot(elec_.R.size(), orbitalsetsize);
+  SPOSet::ValueMatrix d2psiM_rot(elec_.R.size(), orbitalsetsize);
+  rot_spo->evaluate_notranspose(elec_, 0, elec_.R.size(), psiM_rot, dpsiM_rot, d2psiM_rot);
+
+  /* 
+     JPT DEBUG: Ugly but it works: manually encode the unitary transformation.
+     @TODO: Use the total rotation machinery when it's implemented
+
+     NB: This is truncated to 5 sig-figs, so there is some slop here as
+         compared to what is done in the splines via apply_rotation().
+	 So below we reduce the threshold for comparison. This can
+	 probably be ditched once we have a way to grab the actual 
+	 rotation matrix...
+  */
+  SPOSet::ValueMatrix rot_mat(orbitalsetsize, orbitalsetsize);
```

Comment:
Could you copy this part directly into test_einset.cpp and test `SplineR2R::applyRotation`?

Reply:
Done. Extended test_einset to explicitly construct a SplineR2R and call applyRotation(). Now there is a test of the high-level API in test_rotatedspos.cpp, and a test of the low-level implmentation in test_einset.cpp.